azurelinuxagent/common/cgroupconfigurator.py

```diff
@@ -511,3 +431,73 @@ def get_default_memory_limits(cgroup_name):
         if AGENT_CGROUP_NAME.lower() in cgroup_name.lower():
             mem_limit = min(DEFAULT_MEM_LIMIT_MAX_MB, mem_limit)
         return mem_limit
+
+
+class CGroupConfigurator_tmp(object):
+    #
+    # TODO: code from CGroupConfigurator is being moved to this class
+    #
+    class __impl(object):
+        def __init__(self):
+            """
+            Ensures the cgroups file system is mounted and selects the correct API to interact with it
+            """
+            self.cgroups_api = None
+
+            if CGroupConfigurator.enabled():
+                try:
+                    CGroupConfigurator._osutil.mount_cgroups()
+                    self.cgroups_api = SystemdCgroupsApi() if CGroupConfigurator.is_systemd() else FileSystemCgroupsApi()
+                    status = "The cgroup filesystem is ready to use"
+                except Exception as e:
+                    status = ustr(e)
+                    CGroupConfigurator.disable()
+            else:
+                status = "Cgroups are not supported by the platform"
+
+            logger.info("CGroups Status: {0}".format(status))
+
+            add_event(
+                AGENT_NAME,
+                version=CURRENT_VERSION,
+                op=WALAEventOperation.InitializeCGroups,
+                is_success=CGroupConfigurator.enabled(),
+                message=status,
+                log_event=False)
+
+        def _invoke_cgroup_operation(self, operation, error_message):
+            """
+            Ensures the given operation is invoked only if cgroups are enabled and traps any errors on the operation
+            """
+            if not CGroupConfigurator.enabled():
+                return
+
+            try:
+                operation()
```

Comment:
What does this do? Or is this a stub for now?

Reply:
it's the call to the operation passed as argument... lines 488 & 494 are examples of its use

Reply:
Got it.